### DISCUSSION TITLE
Why do RandomLinkSplit and RandomNodeSplit have different types of outputs?

### DISCUSSION THREAD

**Discussion by meredith-martz** [pyg-team/pytorch_geometric#7658] [General]
Is there a reason that RandomNodeSplit adds `train_mask`, `val_mask` and `test_mask` attributes to the Data object whereas RandomLinkSplit returns three different subgraphs? It is confusing as the end user that the calling behavior is different for the two functions. I would expect it to be consistent across the two methods.

**Comment by rusty1s** [comment]
I agree, it‘s sadly not consistent. In general, it is preferred to only have one `data` object with mask attributes (e.g., to share data more easily across different splits). However, this is not possible in random link splits as the graph actually changes across splits. There is an option to only have one `data` attribute in random link splits as well though (see `split_labels` argument).

**Reply by meredith-martz** [reply]
Fair enough!
